core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -95,7 +97,7 @@
 
     private volatile ScheduledFuture nodesSamplerFuture;
 
-    private final AtomicInteger randomNodeGenerator = new AtomicInteger();
+    private final AtomicInteger randomNodeGenerator = new AtomicInteger(ThreadLocalRandom.current().nextInt());
```

Comment:
This will fail the build, `ThreadLocalRandom` is forbidden. You can use `Randomness#get`.